### PR TITLE
feat(ImagePreview): add swipeTo method

### PR DESCRIPTION
**Pull request by laihuamin**
### Before submitting a pull request, please make sure the following is done:

1. Read the [contributing guide](https://github.com/youzan/vant/blob/dev/.github/CONTRIBUTING.md).
2. If you've added code that should be tested, add tests.
3. If you've changed APIs, update the documentation.
4. Ensure the test suite passes (`npm test`).

#### Title Format

type(ComponentName?)：commit message

Example：

- docs: fix typo in quickstart
- build: optimize build speed
- fix(Button): incorrect style
- feat(Button): add color prop

Allowed Types:

- fix
- feat
- docs
- perf
- test
- types
- build
- chore
- refactor
- breaking change

### CODE REVIEW

File: src/image-preview/ImagePreview.js [hunk 1]
```diff
@@ -405,6 +405,10 @@ export default createComponent({
     onClosed() {
       this.$emit('closed');
     },
+
+    swipeTo(index) {
+      this.$refs.swipe.swipeTo(+index, { immediate: true });
+    },
```
Comment (chenjiahan) [inline src/image-preview/ImagePreview.js:411]: 两个参数透传即可，没必要做转换或封装

Comment (chenjiahan) [reply src/image-preview/ImagePreview.js:411]: 增加一个 this.$refs.swipe 的非空判断

File: src/image-preview/demo/index.vue [hunk 2]
```diff
@@ -32,6 +32,25 @@
         <template #index>{{ t('index', index) }}</template>
       </van-image-preview>
     </demo-block>
+
+    <demo-block :title="t('componentPage')">
```
Comment (chenjiahan) [inline src/image-preview/demo/index.vue:36]: 添加示例时，需要补全对应的中英文文档

File: src/image-preview/demo/index.vue [hunk 3]
```diff
@@ -32,6 +32,25 @@
...
+        :images="images"
+        @change="onChange"
+        ref="imagePreview"
+      >
+        <template #index>
+          <div class="block__wrap">
+            <div class="block">{{ t('index', index) }}</div>
+            <div class="block__btn" @click="swipeToSecond">前往第三页</div>
+          </div>
+        </template>
```
Comment (chenjiahan) [inline src/image-preview/demo/index.vue:51]: 示例的内容尽可能简单些

Comment (chenjiahan) [reply src/image-preview/demo/index.vue:51]: 不应该在代码里写死中文，需要使用 i18n

File: src/image-preview/demo/index.vue [hunk 4]
```diff
@@ -53,6 +72,7 @@ export default {
       button3: '异步关闭',
       button4: '展示关闭按钮',
       componentCall: '组件调用',
+      componentPage: '指定滑动位置',
```
Comment (chenjiahan) [inline src/image-preview/demo/index.vue:75]: componentPage 改成 swipeToPosition

File: src/image-preview/ImagePreview.js (hunk 1 above)
Comment (chenjiahan) [reply src/image-preview/ImagePreview.js:411]: 增加一行注释：`// @exposed-api`

Comment (chenjiahan) [reply src/image-preview/ImagePreview.js:411]: 添加对应的中英文 API 文档

Comment (chenjiahan) [reply src/image-preview/ImagePreview.js:411]: 在 types 目录下添加对应的类型定义，可以的话再补个单测

File: src/image-preview/demo/index.vue (hunk 3 above)
Comment (laihuamin) [reply src/image-preview/demo/index.vue:53]: done

File: src/image-preview/ImagePreview.js (hunk 1 above)
Comment (laihuamin) [reply src/image-preview/ImagePreview.js:415]: done

**Pull request merged**